allennlp/fairness/fairness_metrics.py

```diff
@@ -0,0 +1,921 @@
+            }
+            return kl_divs
+
+        C_dist = Categorical(self._predicted_label_counts / self._total_predictions)
+        for a in range(self._num_protected_variable_labels):
+            C_given_a_dist = Categorical(
+                self._predicted_label_counts_by_protected_variable_label[a]
+                / self._total_predictions
+            )
+            kl_divs[a] = kl_divergence(C_given_a_dist, C_dist)
+        if reset:
+            self.reset()
+        return kl_divs
+
+    @overrides
+    def reset(self) -> None:
+        self._predicted_label_counts = torch.zeros(self._num_classes)
+        self._total_predictions = torch.tensor(0)
+        self._predicted_label_counts_by_protected_variable_label = {
+            a: torch.zeros(self._num_classes) for a in range(self._num_protected_variable_labels)
+        }
+
+
+@Metric.register("separation")
+class Separation(Metric):
+    """
+    Separation. Assumes integer labels, with
+    each item to be classified having a single correct class.
+    """
+
+    def __init__(self, num_classes: int, num_protected_variable_labels: int) -> None:
+        """
+        # Parameters
+
+        num_classes : `int`
+            Number of classes.
+        num_protected_variable_labels : `int`
+            Number of protected variable labels.
+        """
+        self._num_classes = num_classes
+        self._num_protected_variable_labels = num_protected_variable_labels
+        self._predicted_label_counts_by_gold_label = {
+            y: torch.zeros(num_classes) for y in range(num_classes)
+        }
+        self._total_predictions = torch.tensor(0)
+        self._predicted_label_counts_by_gold_label_and_protected_variable_label = {
+            y: {a: torch.zeros(num_classes) for a in range(num_protected_variable_labels)}
+            for y in range(num_classes)
+        }
+
+    @overrides
+    def __call__(
+        self,
+        predicted_labels: torch.Tensor,
+        gold_labels: torch.Tensor,
+        protected_variable_labels: torch.Tensor,
+        mask: Optional[torch.BoolTensor] = None,
+    ) -> None:
+        """
+        # Parameters
+
+        predicted_labels : `torch.Tensor`, required.
+            A tensor of predicted integer class labels of shape (batch_size, ...). Represented as C.
+        gold_labels : `torch.Tensor`, required.
+            A tensor of ground-truth integer class labels of shape (batch_size, ...). It must be the same
+            shape as the `predicted_labels` tensor. Represented as Y.
+        protected_variable_labels : `torch.Tensor`, required.
+            A tensor of integer protected variable labels of shape (batch_size, ...). It must be the same
+            shape as the `predicted_labels` tensor. Represented as A.
+        mask : `torch.BoolTensor`, optional (default = `None`).
+            A tensor of the same shape as `predicted_labels`.
+
+            Note: all tensors are expected to be on the same device.
```

Comment:
You could do format like this to make it look better on the docs site:

```suggestion

        !!! Note
            All tensors are expected to be on the same device.
```

Reply:
Woah, this is so cool :D I just changed it everywhere